tests/end2end/test_lpar.py

```diff
@@ -229,3 +234,223 @@ def test_lpar_list_os_messages(classic_mode_cpcs):  # noqa: F811
+        'T28': 'BLUEC1',
+    },
+    'linux-with-autoload': {
+        # Linux profile that does auto-load and specifies a bootable OS
+        'P0000M96': 'HISOCKE1',
+        'T28': 'BCORE2',
+    },
+    'ssc': {
+        # SSC profile (which always auto-loads), that specifies a bootable
+        # SSC image (installer or appliance)
+        'P0000M96': 'ANGEL',
+        'T28': 'BCORE1',
+    },
+}
+
+LPAR_ACTIVATE_TESTCASES = [
+    # Testcases for test_lpar_activate().
+    # The list items are tuples with the following items:
+    # - desc (string): description of the testcase.
+    # - lpar_kind (string): LPAR kind to be used (from ACTIVATE_TEST_LPARS).
+    # - profile_kind (string): Image profile kind to be used (from
+    #   ACTIVATE_TEST_PROFILES). None means to use default (next profile).
+    # - input_kwargs (dict): Input parameters for Lpar.activate() in addition
+    #   to activation_profile_name.
+    # - exp_props (dict): Dict of expected properties of the Lpar after
+    #   calling Lpar.activate().
+    # - exp_exc_type (class): Expected exception type, or None for success.
+    # - run (bool or 'pdb'): Whether to run the test or call the debugger.
+
+    (
+        "Linux LPAR whose same-named profile is w/o auto-load, default parms",
+        'linux-no-autoload',
+        None,
+        dict(),
+        {
+            'status': 'not-operating',
+        },
+        None,
+        True,
+    ),
+    (
+        "Linux LPAR whose same-named profile is with auto-load, default parms",
+        'linux-with-autoload',
+        None,
+        dict(),
+        {
+            'status': 'operating',
+        },
+        None,
+        True,
+    ),
+    (
+        "SSC LPAR (always auto-loads), default parms",
+        'ssc',
+        None,
+        dict(),
+        {
+            'status': 'operating',
+        },
+        None,
+        True,
+    ),
+    (
+        "Linux LPAR, different image profile name than LPAR name specified",
+        'linux-no-autoload',
+        'linux-with-autoload',
+        dict(),
+        None,
+        # HTTP 500,263: LPAR name and image profile name must match
+        zhmcclient.HTTPError,
+        True,
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "desc, lpar_kind, profile_kind, input_kwargs, exp_props, exp_exc_type, run",
+    LPAR_ACTIVATE_TESTCASES)
+def test_lpar_activate(
+        desc, lpar_kind, profile_kind, input_kwargs, exp_props, exp_exc_type,
+        run, classic_mode_cpcs):  # noqa: F811
+    # pylint: disable=redefined-outer-name, unused-argument
+    """
+    Test Lpar.activate().
+    """
+    if not classic_mode_cpcs:
+        pytest.skip("HMC definition does not include any CPCs in classic mode")
+
+    if not run:
+        skip_warn("Testcase is disabled in testcase definition")
+
+    logger = setup_logging(LOGGING, 'test_lpar_activate', LOG_FILE)
+
+    for cpc in classic_mode_cpcs:
+        assert not cpc.dpm_enabled
+
+        # client = cpc.manager.client
```

Comment:
If not needed delete these commented lines

Reply:
DONE